tools/build_info.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#pragma once
+
+#include "build_mode.hh"
+
+namespace tools::build_info {
+enum class build_type { DEBUG = 0, RELEASE, DEV, SANITIZE, COVERAGE, UNKNOWN };
+
+constexpr build_type get_build_type() {
+#ifdef SCYLLA_BUILD_MODE_DEBUG
+    return BuildType::DEBUG;
```

Comment:
btw. from the build errors macro setup it looks like the build can be both DEBUG and SANITIZE, so perhaps the build type might need to be a set of possibly multiple options (COVERAGE could probably also be used together with another build type)

Reply:
It does not look like it, see build_mode.hh: 
```

#if SCYLLA_BUILD_MODE_CODE(SCYLLA_BUILD_MODE) == SCYLLA_BUILD_MODE_CODE_debug
#define SCYLLA_BUILD_MODE_DEBUG
#elif SCYLLA_BUILD_MODE_CODE(SCYLLA_BUILD_MODE) == SCYLLA_BUILD_MODE_CODE_release
#define SCYLLA_BUILD_MODE_RELEASE
#elif SCYLLA_BUILD_MODE_CODE(SCYLLA_BUILD_MODE) == SCYLLA_BUILD_MODE_CODE_dev
#define SCYLLA_BUILD_MODE_DEV
#elif SCYLLA_BUILD_MODE_CODE(SCYLLA_BUILD_MODE) == SCYLLA_BUILD_MODE_CODE_sanitize
#define SCYLLA_BUILD_MODE_SANITIZE
#elif SCYLLA_BUILD_MODE_CODE(SCYLLA_BUILD_MODE) == SCYLLA_BUILD_MODE_CODE_coverage
#define SCYLLA_BUILD_MODE_COVERAGE
#else
#error unrecognized SCYLLA_BUILD_MODE
#endif
```

Reply:
hum ... so it looks it should probably work. I was just wondering as in the build error I saw that both DEBUG and SANITIZE macros were defined as 1.